Add ts7180 support

lionel opened the pull request:
Initial pull request. 

I will start by backing out the last two commits, which are for the ts7100.

randytsi369 committed:
ARM: dtb: ts7180: Initial commit for port to linux-lts kernel repo

Supports Rev C PCBs and later.

Summary roll-up of commits previously in the embeddedts/linux-4.9.y
tree plus commits added during the port to 5.10:

  * SHIP? ARM: dts: ts7180: Add vdd-supply to gyro/accel and magn
  * pull vdd-supply that SUPPOSEDLY should not be needed for gyro/magn to configure

  * ARM: dts: ts7180: Use rts-gpios not uart-tx-enable-gpios; omit fsl (for UARTs)

    The 4.1 kernel didn't have txen support, so we added it using the old
    gpios string. To pick it up now in 4.9, we need rts-gpios intead.

  * ARM: dts: ts7180: fix GPIO name EN_AUX_PWR to EN_CELL_MODEM_PWR

    To match the schematic.

  * ARM: dtb: ts7180: Add Silabs C2 pin iomux definitions
  * ARM: dtb: ts7180: Added PPS GPIO connected to the GPS PPS output.  This requires FPGA revision 15 or greater.
  * ARM: dts: TS: Update FDTs for new WILC driver
  * ARM: dts: ts7180: Corrected node for bit-banged I2C

    - Should be GPIO_ACTIVE_HIGH (high is 1 on i2c), not
      GPIO_ACTIVE_LOW. With I2C, 0 is *driven* low, but writing a 0 bit
      still outputs 0. So it is active high.

    - sda-open-drain and scl-open-drain must not be set, even when
      open-drain is enabled on the GPIO.

    - Disable un-needed internal pull-ups for HD12 I2C

  * ARM: dts: ts7180: Rename HD1 signals split out onto HD12 as of Rev C
  * ARM: dts: ts7180: renames to embeddedTS
  * ARM: dts: ts7180: Disable line driver for GPIO3_IO18 (PUSH_SW_CPU#)

    For some reason, this change enables the push switch to work, even
    though it continues to work if the line driver enabled after boot.

  * ARM: dts: ts7180: Add ts7100-wdt (watchdog) on i2c1

    This is always needed, because otherwise a timeout set in U-Boot will
    not be cancelled and the board will reboot.

    The entry's comment includes a breadcrumb to explain why the timeout
    that is read back from hardware is likely to differ from whatever is
    set here:

    Userspace procs can still start feeding with different values later -
    watchdog(8) defaults to 60 seconds.

    The CPU's builtin watchdog (wdog1) has a trace enabling it to assert
    RST# (analogous to EXT_RESET#/PWR_CYCLE# on the C99). We disable the
    watchdog, but leave the pinmux settings in place.

    wdog1 should be disabled, because otherwise both it and the ts7100-wdt
    will have a 60-second timeout, making it possibly random which
    reset circuitry will fire first. Don't enable wdog1 unless ts7100-wdt
    is disabled or is given a different timeout.

  * ARM: dts: ts7180: rearrange DTS for easier diff w/7250v3

    &cpu0 and &gpc are two interesting ones in the 7180 tree that aren't
    in the 7250v3.

    Neither is in particularly alphabetical order. The remaining i2c
    entries aren't even adjacent to &i2c1.

  * ARM: ts7180: dts: Move the regulators out of their sub-level
  * dts: rename leds to led-controller nodes
  * ARM: dts: ts7180: follow the convention of aliases at head of dtsi else dts
  * ARM: dts: ts7180: update copyright before shipping
  * ARM: dts: ts7180: reserved-memory does not belong in DTS and can break CMA w/7250v3 config
  * memory@80000000 and device_type=memory to match other TS boards
  * ARM: dts: ts7180: pxp_v4l2 does not belong anymore
  * ARM: dts: ts7180: ethernet needs a fixed- rather than dummy-regulator
    The regulator made the difference for ethernet

  * ARM: dts: ts7180: get st_magn with compat string of lis2mdl, not lis2mdl_magn
  * ARM: dts: ts7180: The ism330dlc now handled by the lsm6dsx gyro/accel driver
  * ARM: dts: ts7180: The lis2mdl_magn is (now) handled by the st_magn driver

    - the ISM330 driver initializes the chips as level-high, not edge-triggered
    - fix typo inherited from 7250v3

  * ARM: dts: ts7180: use ts7100-wdt rather than ts-wdt

  * The great alphabeticalization
  * columnize gpio6 (on the FPGA)
  * columnize gpio1 through gpio5, all on the 6ul

  * remove dvfs (not used outside NXP's kernel)
  * remove delay-us for i2c-gpio
  * use fsl,num-cs rather than spi-num-chipselects

  * By default, use the CPU watchdog rather than the SMC watchdog
  * ARM: dts: ts7180: FRAM SPI CLK 20 -> 1 MHz b/c it is shared offboard
  * ARM: dts: ts7180: uart-has-rtscts should never have been on UART with GPIO flow control

lionel commented:
At @markfeathers request, I squashed all device tree commits for the TS-7180 down to a single one.

lionel commented:
I also removed commits into `tsimx6ul_defconfig` and its minimal sibling. That is so to avoid non-essential changes. If what's already here isn't good enough, it'll be added.

ts-kris commented:
> At @markfeathers request, I squashed all device tree commits for the TS-7180 down to a single one.

Yes, for an initial commit like this, it is preferred to have a single commit for each logical thing added. e.g. device trees, drivers, etc., rather than the incremental commits that tend to happen during early dev.


> I also removed commits into tsimx6ul_defconfig and its minimal sibling.

I'd like to see what that diff was, if you still have it. e.g. a config option for supporting the WDT would be wise and shouldn't harm existing configs.


> caddc8326f32411567db8c1678593a978964865d

Unless absolutely necessary, we don't want separate configs for each board. This also is a rename, so it removes our `tsimx6ul_defconfig` file. I'd like to see a proposed commit for changes to `tsimx6ul_[minimal_]defconfig`.

lionel commented:
@ts-kris I'm all for squashing down historical minutiae (that can anyhow be documented in the roll-up commit's full message). Though, the alternative I had in mind was leaving the 4.9 commit(s) in place so that the 5.10 changes would be more apparent.

None of the pre-existing 5.10.y defconfigs should have changed let alone disappeared under this P.R. Looking now to see. I had been keeping a separate ts7180 defconfig so I can figure out what broke in the common defconfigs, especially while they were in flux. I'll definitely delete the `ts7180_defconfig` before this P.R. is merged into 5.10.y.

lionel commented:
Should be no defconfig differences now.

ts-kris commented:
> Though, the alternative I had in mind was leaving the 4.9 commit(s) in place so that the 5.10 changes would be more apparent.

FWIW none of the other FDTs were imported that way, cf 12cad7244ca57b3a4ec9e54f62fea50cd68e8343, b3b9df549ebf994067b05988d5d7b997d18acdf7 They basically were indeed taking that base commit, modifying it to work in 5.10.y, and then smashing it all down just to reduce that noise.


I'd still also like to see the defconfigs get updated to have support for the WDT that was added.

Otherwise, looks like its mostly ready for review? I'll take a look now.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ *
+ */
```
Use kernel standard license text. Other platforms we've added use the template:

```
// SPDX-License-Identifier: GPL-2.0 OR X11
/*
 * Copyright yyyy-zzzz Technologic Systems, Inc. dba embeddedTS
 */
```

The `OR X11` seems to be pretty common practice. Been a while since I dug in to it and I forget exactly why that was added.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+#include "imx6ul.dtsi"
+
+/ {
+	model = "embeddedTS i.MX6ul TS-7180 (G)";
```
`i.MX6UL`

What is `(G)`?

lionel replied:
A marker I put in so I could easily see which device tree was booted :-) Will remove.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+
+	led-controller {
+		compatible = "gpio-leds";
+		pinctrl-names = "default";
```
Remove if no actual `pinctrl` is specified for the LEDs. If there is a `pinctrl` then `pinctrl0 = <&...>;` is missing here (which it looks like it might be since these are all `<&gpio4>`

lionel replied:
Changed this to follow the example, e.g., of the TS-4900.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+
+		led-0 {
+			color = <LED_COLOR_ID_YELLOW>;
+			label = "yellow-led";
```
`label` property is deprecated, `color` property takes care of that.

Add `function` property as well for each LED.

In general we use:
`LED_FUNCTION_POWER` for green LED
`LED_FUNCTION_STATUS` for red LED
`LED_FUNCTION_INDICATOR` for all other colors. (INDICATOR is not _preferred_ by kernel, but none of the preferred FUNCTIONS make sense with how we don't use these LEDs out of the box)

lionel replied:
I made this change. I couldn't tell from our examples nor [this kernel document](https://www.kernel.org/doc/Documentation/devicetree/bindings/leds/common.yaml) whether the # in "`led-# {}`" was meant to correspond to anything specific. I used the LED# from the schematic and sorted the LEDs in order of #.

ts-kris replied:
> I couldn't tell from our examples nor this kernel document whether the # in "led-# {}" was meant to correspond to anything specific.

No, its just a number. I think all of the other DTSs we've done have the same or similar order, I think usually green/red/yellow/blue.


> I used the LED# from the schematic and sorted the LEDs in order of #.

That's fine, as at runtime the color and function determine their name anyway.

lionel commented:
@ts-kris, @markfeathers and I discussed backing out my `ts7100_wdt.c` that works with the 7180 and just pulling forward whatever changes are truly needed in 5.10. So no need to wrack your brain on the WDT quite yet.

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+/*
+ * Copyright (C) 2021-2023 Technologic Systems, Inc. dba embeddedTS
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ *
+ */
```
```suggestion
// SPDX-License-Identifier: GPL-2.0 OR X11
/*
 * Copyright 2017-2023 Technologic Systems, Inc. dba embeddedTS
 */
```

We do not want our stock device trees to be GPL only

lionel replied:
Fixed per earlier comment re: SPDX.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+			label = "en-sd";
+			gpio = <&gpio6 14 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+		};
```
Remove these if possible (that is, remove them unless there is literally no other way to get the board booting), they are not correct and we should no longer be abusing them.

The bootloader should be setting the default state, so the kernel would not need to specify it. Same with the output enable bits of these pins. Ideally, SD/eMMC would be under regulator control. However, since we separately want eMMC power control for `mmc` commands (e.g. to enable high reliability modes), the eMMC power enable can remain as a named GPIO.

lionel replied:
Removed pending testing of SD-card boot.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+		};
+	};
+
+		reg_gpio_dvfs: regulator-gpio {
```
All regulator names should follow `reg_<peripheral>_<function>...: regulator-<peripheral>` (or close to it, see other imx6ul-ts* dts files for examples).

lionel replied:
OK; However, I removed this, per my message in engineering chat.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+			pinctrl-0 = <&pinctrl_dvfs>;
+			regulator-min-microvolt = <1300000>;
+			regulator-max-microvolt = <1400000>;
+			regulator-name = "gpio_dvfs";
```
Try to make names all caps, we've been standardizing on all caps for regulator names and GPIO names.

lionel replied:
Done.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+			regulator-name = "ADC_VREF";
+			regulator-min-microvolt = <2500000>;
+			regulator-max-microvolt = <2500000>;
+			//gpio = EN_2.5_VREF driven by Silabs;
```
If this is driven by uC, then this should be named as a fixed voltage regulator. We don't care what its for, its up to the peripheral (the ADC driver in this case) to call it out.

e.g.
```

        reg_1v8: regulator-1v8 {
                compatible = "regulator-fixed";
                regulator-name = "1V8";
                regulator-min-microvolt = <1800000>;
                regulator-max-microvolt = <1800000>;
                regulator-always-on;
        }
```

lionel replied:
Fixed.

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+#include "imx6ul.dtsi"
+
+/ {
+	model = "embeddedTS i.MX6ul TS-7180 (G)";
```
```suggestion
	model = "embeddedTS i.MX6ul TS-7180";
```

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+
+	led-controller {
+		compatible = "gpio-leds";
+		pinctrl-names = "default";
```
The LEDs should be in their own pinctrl group rather than the hog

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+			//enable-active-high;
+		};
+
+		reg_flexcan_3v3: en-can@0 {
```
Don't need the @0 unless we have multiple can enable pins

lionel replied:
Fixed.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+			gpio = <&gpio6 20 GPIO_ACTIVE_HIGH>;
+		};
+
+	reg_3v3: regulator-3v3 {
```
Fix indenting

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-always-on;
+		pinctrl-0 = <>; /* still needed if always-on? */
```
No it is not needed.

lionel replied:
Thanks!-)

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+			color = <LED_COLOR_ID_BLUE>;
+			label = "blue-led";
+			gpios = <&gpio4 26 GPIO_ACTIVE_HIGH>;
+			default-state = "off";
```
The LEDs should not have labels for colors since Linux has a standard naming based on the function and specified color id. See the function examples [from the TS-7970 ](https://github.com/embeddedTS/linux-lts/blob/linux-5.10.y/arch/arm/boot/dts/imx6qdl-ts7970.dtsi#L20) as an example.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+		pinctrl-0 = <>; /* still needed if always-on? */
+	};
+
+        i2c_gpio@0 {
```
Should be `i2c_gpio: i2c {` Needs the label and node name, doesn't need @ unless we have multiple clashing names.

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+			label = "en-sd";
+			gpio = <&gpio6 14 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+		};
```
These should be removed and the pin should have a pull to the enable direction. This should be in the appropriate pinctrl group for the emmc/SD so its turned on with everything else in the group. The IO would otherwise be discovered through gpiofind with an appropriate label. I would like to have a regulator except we have no way to toggle it in userspace once claimed by this driver, and we need that for emmc.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+	        clock-frequency = <100000>;
+	        pinctrl-names = "default";
+	        pinctrl-0 = <&pinctrl_i2c_gpio>;
+		gpios = <
+			&gpio3 9 GPIO_ACTIVE_HIGH /* I2C_DAT on HD12 pin 1 */
+			&gpio3 8 GPIO_ACTIVE_HIGH /* I2C_CLK on HD12 pin 6 */
+			>;
```
Preferred properties for these are, e.g.
```
                sda-gpios = <&gpio5 5 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN)>;
                scl-gpios = <&gpio5 4 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN)>;
```

lionel replied:
Missed that one. Thanks.

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+			regulator-min-microvolt = <2500000>;
+			regulator-max-microvolt = <2500000>;
+			//gpio = EN_2.5_VREF driven by Silabs;
+			//enable-active-high;
```
```suggestion
```

lionel replied:
Removed.

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+			compatible = "regulator-fixed";
+			regulator-name = "en-can";
+			regulator-min-microvolt = <3000000>;
+			regulator-max-microvolt = <3000000>;
```
```suggestion
			regulator-min-microvolt = <3300000>;
			regulator-max-microvolt = <3300000>;
```

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+
+        i2c_gpio@0 {
+	        compatible = "i2c-gpio";
+	        clock-frequency = <100000>;
```
Either `clock-freqency` or `delay-us` are needed, not both, and I'm not sure which is preferred.

lionel replied:
The other boards using `i2c-gpio` aren't supplying either. Regardless, none of our boards are using `delay-us`, so I removed that.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+			&gpio3 9 GPIO_ACTIVE_HIGH /* I2C_DAT on HD12 pin 1 */
+			&gpio3 8 GPIO_ACTIVE_HIGH /* I2C_CLK on HD12 pin 6 */
+			>;
+	        status = "okay";
```
Not needed, default status is `okay`

lionel replied:
Fixed.

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-always-on;
+		pinctrl-0 = <>; /* still needed if always-on? */
```
```suggestion
```
pinctrl-0 only needs to be set if we're specifying a group of pins, and then we would need to specify pinctrl-names if we are specifying it. Since there is no pin to control this both should be omitted here

lionel replied:
Removed.

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+		gpios = <
+			&gpio3 9 GPIO_ACTIVE_HIGH /* I2C_DAT on HD12 pin 1 */
+			&gpio3 8 GPIO_ACTIVE_HIGH /* I2C_CLK on HD12 pin 6 */
+			>;
```
This changed a bit with the newer kernel. gpios should be removed and replaced with:
```suggestion
		scl-gpios = <&gpio3 8 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
		sda-gpios = <&gpio3 9 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
```

lionel replied:
Done.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+        };
+
+	pps {
+		compatible = "pps-gpio";
```
Also, need to make sure this driver is present in both defconfigs, I am not sure if it is or not but I don't think so as I think this is the first 6ul we've rolled this out on.

lionel replied:
Added `CONFIG_PPS=m` to `tsimx6ul_defconfig` and `tsimx6ul_minimal_defconfig`.

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+		pinctrl-0 = <&pinctrl_ppsgpio>;
+		gpios = <&gpio5 1 GPIO_ACTIVE_HIGH>;
+
+		status = "okay";
```
```suggestion
```

This status can be removed. Since this is not referencing a node that is disabled like they often are in the upper dtsi files, it is by default "okay" when it is specified.

ts-kris replied:
Not needed, default status is `okay`

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+			&gpio3 9 GPIO_ACTIVE_HIGH /* I2C_DAT on HD12 pin 1 */
+			&gpio3 8 GPIO_ACTIVE_HIGH /* I2C_CLK on HD12 pin 6 */
+			>;
+	        status = "okay";
```
```suggestion
```

lionel replied:
Removed.

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+		interrupt-parent = <&gpio4>;
+		interrupts = <10 IRQ_TYPE_LEVEL_HIGH>;    /* See MX6UL_PAD_NAND_ALE__GPIO4_IO10 */
+		vdd-supply = <&reg_3v3>;
+		vddio-supply = <&reg_3v3>;
```
I'm not finding anything in the driver that benefits from a vdd-supply or vddio-supply. Are these used for anything? If not they should likely be removed.

lionel replied:
The gyro and magnetometer weren't being loaded without these. I need to check to see if that changed in the latest 5.10.y.

lionel replied:
I'll remove these If IIO checks out OK without them.

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+		interrupts = <10 IRQ_TYPE_LEVEL_HIGH>;    /* See MX6UL_PAD_NAND_ALE__GPIO4_IO10 */
+		vdd-supply = <&reg_3v3>;
+		vddio-supply = <&reg_3v3>;
+		status = "okay";
```
```suggestion
```

ts-kris replied:
Not needed, default status is `okay`

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+		reg = <0x6a>;
+		interrupt-parent = <&gpio4>;
+		interrupts = <20 IRQ_TYPE_LEVEL_HIGH>;    /* See MX6UL_PAD_CSI_HSYNC__GPIO4_IO20 */
+		vdd-supply = <&reg_3v3>;
```
Same here, if this is not used for anything this should likely be removed.

lionel replied:
Will handle this gyro instance whichever way the magnetometer instance pans out.

lionel replied:
I'm going to mark this resolved so you can start reviewing the latest updates, but will test and remove this if possible as part of checking IIO.

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+		 */
+		timeout-sec = <300>;
+		reg = <0x54>;
+                status = "okay";
```
```suggestion
```

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_adc>;
+	vref-supply = <&reg_adc_vref>;
+	num-channels = <10>;
```
```suggestion
```
I can't find anything using this in the driver, this might be a remnant from the nxp fork? If so this should be removed.

lionel replied:
Removed.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+	 *
+	 * drdy-int-pin is mandatory, not optional, when using interrupts.
+	 */
+	magn: magnetometer@1e {
```
Format should be `<device_part_num-or-driver_name>: magnetometer@1e {`

lionel replied:
An example? Our other boards look like they use "magnet" or nothing.

lionel replied:
I went with `st_magn` to match the driver name.

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+	arm-supply = <&reg_arm>;
+	soc-supply = <&reg_soc>;
+	dc-supply = <&reg_gpio_dvfs>;
+};
```
This should be removed. The dc-supply is not used by the upstream node and the remaining values are set in imx6ul.dtsi.

lionel replied:
Removed.

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+
+&ecspi3 {
+	fsl,spi-num-chipselects = <3>;
+	cs-gpios = <&gpio4 12 GPIO_ACTIVE_HIGH>, <&gpio3 0 GPIO_ACTIVE_HIGH>, <&gpio4 27 GPIO_ACTIVE_HIGH>;
```
At some point between 4.9 and 5.10, they started using the real polarity on these cs gpio.
```suggestion
	cs-gpios = <&gpio4 12 GPIO_ACTIVE_LOW>, <&gpio3 0 GPIO_ACTIVE_LOW>, <&gpio4 27 GPIO_ACTIVE_LOW>;
```

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+	spifram: spi@2 {
+		compatible = "atmel,at25", "cypress,fm25l16b";
+		reg = <2>;
+		spi-max-frequency = <20000000>;
```
We talked about this one offline, but this should be much, much lower since this has offboard stubs and no buffer. Can you make this match the value that was tested in production?

lionel replied:
Yeah, need to pull in that commit. Thanks.

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+		size = <0x800>;
+		address-width = <16>;
+		pagesize = <64>;
+		status = "okay";
```
```suggestion
```

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+
+&ecspi4 {
+	fsl,spi-num-chipselects = <1>;
+	cs-gpios = <&gpio4 9 0>;
```
```suggestion
	cs-gpios = <&gpio4 9 GPIO_ACTIVE_LOW>;
```

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+	 *
+	 *	- GYRO_INT into CSI_HSYNC pin F3
+	 */
+	gyro: ism330@6a {
```
Format should be `ism330: gyrometer@6a {`

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+		interrupt-parent = <&gpio4>;
+		interrupts = <20 IRQ_TYPE_LEVEL_HIGH>;    /* See MX6UL_PAD_CSI_HSYNC__GPIO4_IO20 */
+		vdd-supply = <&reg_3v3>;
+		status = "okay";
```
Not needed.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+		status = "okay";
+	};
+
+	rtc: m41t00s@68 {
```
`m41t00s: rtc@68 {`

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+	};
+
+	rtc: m41t00s@68 {
+		compatible = "m41t00";
```
`st,m41t00`

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+		reset-gpios = <&gpio6 0 0>;     /* RESET,  GPIO 160 */
+		chip_en-gpios = <&gpio6 1 0>;   /* ENABLE, GPIO 161 */
+		interrupt-parent = <&gpio5>;    /* IRQ,    GPIO 136 */
+		interrupts = <8 0>;
```
```suggestion
		reset-gpios = <&gpio6 0 GPIO_ACTIVE_HIGH>;
		chip_en-gpios = <&gpio6 1 GPIO_ACTIVE_HIGH>;
		interrupt-parent = <&gpio5>;
		interrupts = <8 IRQ_TYPE_EDGE_FALLING>;
```

I would remove the numbers from the comments since this flat numberspace and its order are not guaranteed going forward.

lionel replied:
Removed.

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+	fsl,cpu_pdnscr_iso = <0x1>;
+	fsl,ldo-bypass = <0>; /* DCDC, ldo-enable */
+};
+
```
These options are not present outside of the NXP fork, this whole &gpc node can be removed.
```suggestion
```

lionel replied:
Removed.

lionel replied:
Removed

lionel replied:
Won't let me resolve, even though I removed it.

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+	status = "okay";
+};
+
+
```
```suggestion
```

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+		reg = <0x68>;
+	};
+
+	supervisor: supervisor@54 {
```
`supervisor: wdt@54 {`

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+	pinctrl-1 = <&pinctrl_i2c3_gpio>;
+	status = "okay";
+	scl-gpios = <&gpio3 6 GPIO_ACTIVE_HIGH>;
+	sda-gpios = <&gpio3 5 GPIO_ACTIVE_HIGH>;
```
```suggestion
	scl-gpios = <&gpio3 6 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
	sda-gpios = <&gpio3 5 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
```

lionel replied:
Thanks. Missed that one.

lionel replied:
fixed.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+	status = "okay";
+};
+
+&cpu0 {
+	arm-supply = <&reg_arm>;
+	soc-supply = <&reg_soc>;
+	dc-supply = <&reg_gpio_dvfs>;
+};
```
Check these, I do not think upstream kernel uses these. I think this was only in NXP specific kernels trees.

lionel replied:
Removed.

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+
+&snvs_poweroff {
+	status = "okay";
+};
```
We have not been enabling this on our other 6uls. What does this do when we enable it? Does it work as expected even though we don't have vbatt connected?

lionel replied:
Removed.

lionel replied:
removed

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+};
+
+&ecspi3 {
+	fsl,spi-num-chipselects = <3>;
```
`num-cs = ...`

lionel replied:
Fixed. FYI, `imx6qdl-ts7990.dtsi` is still using `api-num-chipselects`.

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+&wdog1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_wdog>;
+	fsl,wdog_b;
```
This might have been from the NXP fork since I'm not seeing it in the 5.10 tree. To toggle the WDOG# signal you likely want this:

```suggestion
	fsl,ext-reset-output;
```

lionel replied:
fixed

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+	};
+
+	/* This is actually the FRAM, which is compatible with the AT25 SPI EEPROM */
+	spifram: spi@2 {
```
`fm25l16b: eeprom@2 {`

lionel replied:
Fixed.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+
+	/* This is actually the FRAM, which is compatible with the AT25 SPI EEPROM */
+	spifram: spi@2 {
+		compatible = "atmel,at25", "cypress,fm25l16b";
```
just `atmel,at25` the cypress device callout isn't preferred as it does not have an associated driver

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+		size = <0x800>;
+		address-width = <16>;
+		pagesize = <64>;
+		status = "okay";
```
Not needed.

lionel replied:
Removed.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+};
+
+&ecspi4 {
+	fsl,spi-num-chipselects = <1>;
```
`num-cs = ...`

lionel replied:
Fixed.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+		chip_en-gpios = <&gpio6 1 0>;   /* ENABLE, GPIO 161 */
+		interrupt-parent = <&gpio5>;    /* IRQ,    GPIO 136 */
+		interrupts = <8 0>;
+		status = "okay";
```
Not needed

lionel replied:
Not sure if I applied this right. See the next review cycle.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+	pinctrl-0 = <&pinctrl_ecspi4>;
+	status = "okay";
+
+	wilc_spi@0 {
```
`wilc: wifi@0 {`

lionel replied:
Most boards are using `wilc3000: wifi@0`, so that is what I went with.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+	status = "okay";
+
+	wilc_spi@0 {
+		compatible = "microchip,wilc1000", "microchip,wilc3000";
```
just `microchip,wilc3000` as the 1000 driver is not actually compatible.

lionel replied:
Fixed.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+		compatible = "microchip,wilc1000", "microchip,wilc3000";
+		reg = <0>;
+		spi-max-frequency = <48000000>;
+		reset-gpios = <&gpio6 0 0>;     /* RESET,  GPIO 160 */
```
Don't use magic numbers, use `GPIO_ACTIVE_...`

lionel replied:
Fixed.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+		reg = <0>;
+		spi-max-frequency = <48000000>;
+		reset-gpios = <&gpio6 0 0>;     /* RESET,  GPIO 160 */
+		chip_en-gpios = <&gpio6 1 0>;   /* ENABLE, GPIO 161 */
```
ditto

lionel replied:
Replaced magic numbers. Poof !-)

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+		reset-gpios = <&gpio6 0 0>;     /* RESET,  GPIO 160 */
+		chip_en-gpios = <&gpio6 1 0>;   /* ENABLE, GPIO 161 */
+		interrupt-parent = <&gpio5>;    /* IRQ,    GPIO 136 */
+		interrupts = <8 0>;
```
Don't use magic numbers, use `IRQ_TYPE...`

lionel replied:
Fixed.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+};
+
+&gpio1 {
+	gpio-line-names =
```
Smash these to be on as few lines as possible. While having them laid out is visually appealing, these are rarely referenced as actual numbers.

This goes for all `&gpio` entries.

lionel replied:
Smashed. As few lines as possible, but not moreso ?-)

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+&gpio1 {
+	gpio-line-names =
+		"BOOT_MODE_0",
+		"NA",
```
If a pin is unused, name it `"",`

lionel replied:
Renamed all instances.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+		"JTAG_FPGA_TCK",
+		"JTAG_FPGA_TMS",
+		"JTAG_FPGA_TDI",
+		"WDOG#",
```
Can this even be used as a GPIO pin?

Be sure all of these GPIO entries only call out pins that can be used as a GPIO. Regulators, peripherals (CAN, I2C, UART, etc.), etc., all label the pins that are used if they use external pins. These show up in a different column in the `/sys/kernel/debug/gpio` file than the GPIO names and make an important distinction in their use.

On platforms like the TS-7180 where there is no reasonable way to make a UART pin an external GPIO, don't give it a name.

lionel replied:
OK, I generally removed everything that didn't have precedents in other products.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+		"NA";
+};
+
+&gpc {
+	fsl,cpu_pupscr_sw2iso = <0x1>;
+	fsl,cpu_pupscr_sw = <0x0>;
+	fsl,cpu_pdnscr_iso2sw = <0x1>;
+	fsl,cpu_pdnscr_iso = <0x1>;
+	fsl,ldo-bypass = <0>; /* DCDC, ldo-enable */
+};
```
Check that the kernel supports these. I think these are only used in NXP specific trees and vanilla kernel doesn't use them.

lionel replied:
Removed.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+	pinctrl-0 = <&pinctrl_uart5>;
+	status = "okay";
+};
+
```
Extra newline

lionel replied:
fixed

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+	status = "okay";
+};
+
+&usdhc1 {
```
Is the SD card properly wired for different UHS speed modes? This was a problem for sure on other 6ul platforms. Be sure these are supported before calling these out.

lionel replied:
Needs investigation.

lionel replied:
Will test and investigate as needed.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+	scl-gpios = <&gpio3 6 GPIO_ACTIVE_HIGH>;
+	sda-gpios = <&gpio3 5 GPIO_ACTIVE_HIGH>;
+
+	gpio6: gpio@28 {
```
`gpio6: gpio-controller@28 {`

lionel replied:
fixed

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+		reg = <0x28>;
+		#gpio-cells = <2>;
+		gpio-controller;
+		base = <160>;
```
I believe `base` is deprecated. I think it should be removed, but I'm not 100% sure on that.

lionel replied:
Needs investigation

lionel replied:
All other ts4900-gpio boards still use it.

markfeathers replied:
Reopening this, base should be removed.

lionel replied:
Got it: "base" is no longer needed nor accepted.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+		base = <160>;
+		ngpios = <64>;
+
+		gpio-line-names =
```
Collapse these to be on as few lines as possible.

lionel replied:
Done.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+			/* Outputs 0-31 */
+			"WIFI_RESET#",
+			"EN_WIFI_PWR",
+			"NA",
```
IOs without connection or valid use should be `"",`

lionel replied:
Fixed.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+			"NA",
+			"NA",
+			"NA",
+			/* Inputs 32-63 */
```
I'm not sure this separation is needed. But if you prefer to keep it, when you collapse the line names, keep this on its own line separating the blocks.

lionel replied:
OK

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+	};
+};
+
+&snvs_poweroff {
+	status = "okay";
+};
```
Is the poweroff input actually used on this platform?

lionel replied:
There's a button that wakes from sleep, but otherwise no. The TS-7553v2 has something possibly similar, `snvs_pwrkey`, about which it says "Unused, enabled for backward compatibility with `ts7553v2-utils`. I can't tell how those utils are using it. I'm inclined to remove it, but will check further up the device tree to be sure.

ts-kris replied:
> There's a button that wakes from sleep, but otherwise no.

Is that a real physical button on the TS-7180 that connects to the SNVS wake (or whatever the CPU function is on the CPU)? If so, then its worth keeping as that could likely be used to wake from sleep modes. But if its otherwise an NC and there is no way to get a signal to it, then we should disable this.


> "Unused, enabled for backward compatibility with ts7553v2-utils. I can't tell how those utils are using it.

It has to do with `/dev/input/event` on the TS-7553-V2. The stock kernel had `snvs_pwrkey` enabled when it shouldn't, and then all of the `lcd` demo utils were set up to be hard-coded on a specific `/dev/input/event` file to get events from the keypad. Removing `snvs_pwrkey` shifts the event numbers and breaks those utils.

Future TS-7553-V2 development, I'm hoping to support the LCD panel in kernel and then deprecating the demos we have since at that point the LCD and keypad would be fully supported by DRM and generic input systems.

lionel replied:
Removed.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+	status = "disabled";
+};
+
+&wdog1 {
```
Try and keep the `&peripheral {` sections in alphabetical order. I wasn't paying attention to this until now.

lionel replied:
fixed

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_wdog>;
+	fsl,wdog_b;
+	status = "disabled";
```
If this is disabled with no chance of us using it, then we just need the `status = "disabled";` property, the rest of the properties can be deleted.

lionel replied:
Leaving the door open for now.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+			>;
+		};
+
+		pinctrl_ppsgpio: ppsgpio {
```
All pinctrl blocks should be in the format of `pinctrl_<useful label>: <useful node>grp{`

lionel replied:
OK. Changed. FYI, the `imx6ul-ts7100-ts8100.dtsi` looks the same as the 7180 did.

lionel replied:
fixed all instances (I hope)

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,1020 @@
+			>;
+		};
+
+		pinctrl_sai2: sai2grp {
+			fsl,pins = <
+				MX6UL_PAD_JTAG_TDI__SAI2_TX_BCLK	0x17088
+				MX6UL_PAD_JTAG_TDO__SAI2_TX_SYNC	0x17088
+				MX6UL_PAD_JTAG_TRST_B__SAI2_TX_DATA	0x11088
+				MX6UL_PAD_JTAG_TCK__SAI2_RX_DATA	0x11088
+				MX6UL_PAD_JTAG_TMS__SAI2_MCLK		0x17088
+			>;
+		};
```
Is there ever a chance we'll use the SAI interface?

lionel replied:
Probably not. Does the TS-4100 use it (see `imx6ul-ts4100.dtsi`)?

lionel replied:
removed.

ts-kris reviewed (changes requested):
See comments. Note that part of the purpose of this `linux-lts` tree is to stage things to be able to easily upstream them. I'm picky about the FDT file because lkml is picky about the same things.

Also, take a look at the output of `make dtbs_check` as this will give you a good amount of information as to how you should format the FDT source. (Note that, I _think_ this command is broken in our 5.10.y tree at the moment. Merging in upstream 5.10.y might fix that.)

WDT driver not reviewed per request.

lionel committed:
ARM: configs: tsimx6ul: Add CONFIG_PPS=m for models with GPS/PPS

lionel committed:
ARM: configs: tsimx6ul_minimal: GPS/PPS models need CONFIG_PPS=m

lionel commented:
I think I got most of the comments except for the alphabeticalization. Anyway, I'll check back Monday night or Tuesday.

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,788 @@
+	 *	- as well as the FPGA
+	 *
+	 * drdy-int-pin is mandatory, not optional, when using interrupts.
+	 */
```
I think the comments on the 5.10 port themselves should not be in this code. This might be confusing to a user who isn't aware of the perspective of the previous 4.9 port with ST's non-upstream drivers. Even the drdy-int-pin comment likely belongs in ST's own documentation, otherwise during our next kernel port we have to maintain this comment and make sure its still true.
```suggestion
```

lionel replied:
Chopped this down.

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,788 @@
+	 * ism330dlc is (now) handled by the lsm6dsx gyro/accel driver
+	 *
+	 *	- GYRO_INT into CSI_HSYNC pin F3
+	 */
```
```suggestion
```

Same here

lionel replied:
Same resolution as above.

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,788 @@
+	                  "",               "",               "",               "";               
+};
+
+&i2c1 {
```
Make sure the I2C devices defined here are in order of their i2c address.

lionel replied:
Now ordered according to address.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,778 @@
+#include "imx6ul.dtsi"
+
+/ {
+	model = "embeddedTS i.MX6ul TS-7180";
```
"i.MX6UL"

lionel replied:
Fixed.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,778 @@
+	model = "embeddedTS i.MX6ul TS-7180";
+	compatible = "technologic,ts7180", "fsl,imx6ul";
+
+	aliases {
+		ethernet0 = &fec2;
+		ethernet1 = &fec1;
+	};
```
Does this actually do the correct ordering? I feel like I tried this on other platforms and it still kept the reversed numbers.

lionel replied:
The ordering does come out correct. These lines might be un-necessary, or might require other changes in concert in order to remove. But in the meanwhile, the documentation, U-Boot and everything seem to be sufficiently in agreement that I'm leaving this alone for now and fixing later if investigation turns up a problem.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,778 @@
+
+	memory@80000000 {
+                device_type = "memory";
+		reg = <0x80000000 0x20000000>;
```
Since this can be configured with two sizes of RAM, the length should not be specified and would be filled in by U-Boot. Common notation for this is:
```
                /* Memory size to be filled in by U-Boot */
                reg = <0x80000000 0>;
```

lionel replied:
Will fix.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,778 @@
+		regulator-always-on;
+	};
+
+        i2c_gpio@0 {
```
Should be `i2c_gpio: i2c {` Needs the label and node name, doesn't need @ unless we have multiple clashing nodes.

lionel replied:
Will fix.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,778 @@
+};
+
+&ecspi3 {
+	fsl,num-cs = <3>;
```
```suggestion
	num-cs = <3>;
```

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,778 @@
+};
+
+&ecspi4 {
+	fsl,num-cs = <1>;
```
`num-cs` as above

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,778 @@
+	spioffbd: spi@1 {
+		compatible = "technologic,spi-header";
+		reg = <1>;
+		spi-cs-high;
```
Is this CS actually inverted?

lionel replied:
No reports of issues, but I'll investigate and fix after this P.R. is complete, since any software written to this behavior would break.

markfeathers replied:
Unresolving this because I think this is still a concern.  Is this actually an active high chip select?  If we aren't inverting this before bringing it to a header, this should be removed.

lionel replied:
4.9 ignored and "did the right thing" in order to work. With 5.10, this will break if wrong, and has already been fixed on other products ported to 5.10. Will change this here.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,778 @@
+};
+
+&gpio1 {
+	gpio-line-names = "BOOT_MODE_0", "",            "I2C_1_CLK",   "I2C_1_DAT",   
```
I like the tab aligned, no clue why I didn't think of that when porting.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,778 @@
+	                  "",            "",            "",            "";
+};
+
+&gpio2 {
+	gpio-line-names = "", "", "", "", "", "", "", "", 
+	                  "", "", "", "", "", "", "", "", 
+	                  "", "", "", "", "", "", "", "", 
+	                  "", "", "", "", "", "", "", ""; 
+};
```
If there are no line names, then this can just be removed.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,778 @@
+	sda-gpios = <&gpio1 3 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	status = "okay";
+
+	/*
+         * Reference: Documentation/devicetree/bindings/iio/st-sensors.txt
+	 *
+	 *   - MAG_N_IRQ enters NAND_ALE/SD2_RST#/PWM3 pad B4
+	 *   - as well as the FPGA
+	 */
```
Remove this, not necessary. The reference can be found by looking at the compatible name. If the FPGA pin is unused and otherwise NC then it doesn't need to be mentioned. And the pin itself can be looked up via the pinctrl table.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,778 @@
+		reg = <0x68>;
+	};
+
+	/*
+	 * Reference: Documentation/devicetree/bindings/iio/imu/st_lsm6dsx.txt
+	 *
+	 *   - GYRO_INT enters CSI_HSYNC (pad F3)
+	 */
```
Remove this, not necessary. The reference can be found by looking at the compatible name. The pin itself can be looked up via the pinctrl table.

lionel replied:
Removed.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,778 @@
+	scl-gpios = <&gpio3 6 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio3 5 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+
+	gpio6: gpio@28 {
```
```suggestion
	gpio6: gpio-controller@28 {
```

lionel replied:
Fixed.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,778 @@
+				MX6UL_PAD_LCD_HSYNC__GPIO3_IO02		0x1b020 /* JTAG_FPGA_TMS */
+				MX6UL_PAD_LCD_ENABLE__GPIO3_IO01	0x1b020 /* JTAG_FPGA_TCK */
+				MX6UL_PAD_CSI_DATA00__GPIO4_IO21	0x10020 /* 6UL_FORCE_5V_ON */
+				/* EN_YEL_LED# in pinctrl_gpio_leds */
+				/* EN_RED_LED# in pinctrl_gpio_leds */
+				/* EN_GREEN_LED# in pinctrl_gpio_leds  */
+				/* EN_BLUE_LED in pinctrl_gpio_leds  */
```
Not necessary

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,778 @@
+	};
+};
+
+&iomuxc {
```
For uniformity sake, try to have the IOMUX values tab aligned through the whole pinmux section. It looks like below some have tabs, spaces, etc. Just so long as its something uniform in the file that's fine.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,778 @@
+			>;
+		};
+
+		pinctrl_pwm5: ts7180pwm5 {
```
Node name should end in grp

lionel replied:
Fixed. Note: `imx6ul-ts4100.dtsi` is also missing this.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,778 @@
+			>;
+		};
+
+		pinctrl_i2c_gpio: i2cgrpgpio {
```
Node name should end in grp

lionel replied:
Any examples of it being done right? The 4100 and 7250v3 do it the same way.

lionel replied:
Fixed for the ts7180.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,778 @@
+	uart-has-rtscts;
+};
+
+&uart4 {
```
Is DMA still an issue with RS-485 or has that been resolved in 5.10.y? If DMA is still an issue, this needs a property added to disable it for this port:
```
dma-names = "", "";
```

lionel replied:
It would need research and testing. At various points, I recall NXP fixes for SDMA when handling things like TXEN, but not sure this was resolved for our situation.

lionel replied:
Disabled until such time as this can be properly tested.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,778 @@
+	pinctrl-0 = <&pinctrl_uart4>;
+	status = "okay";
+	uart-has-rtscts;
+	rts-gpios = <&gpio1 19 GPIO_ACTIVE_HIGH>; /* EN_485 is the TX enable signal */
```
```suggestion
	rts-gpios = <&gpio1 19 GPIO_ACTIVE_HIGH>;
```

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,778 @@
+	status = "okay";
+};
+
+&usdhc1 {
```
Looking at the rev C schematic, UHS modes are not likely to work. Have these been tested?

Sheet 12 has a VSELECT mux, however, the microSD card socket on sheet 10 has the pullups to SD_3.3V. The SD card needs VDD always connected to SD_3.3V, but the IO pullups need to be moved to 6UL_SD_POWER so in 1.8 V UHS modes the pins arn't over-pulled. This needs to be corrected in hardware.

lionel replied:
Needs research. We're likely going to disable UHS until such time as it is resolved in hardware.

lionel replied:
Rev D hardware implements the pull-ups to `6UL_SD_POWER`. Via `dd`, I saw read speeds of 37.7 MB/s under both 4.9 and 5.10.202, which (per @markfeathers) suggests we've been running it "high-speed" for quite some time, even on rev C hardware where we shouldn't have been. Rev D has been shipping for quite some time. So, other than stylistic changes for 5.10, I think we should leave these parameters un-changed for now.

lionel committed:
ARM: dts: ts7180: cleanup style post-debug/pre-alphabeticalization

  - Remove comment: /* EN_485 is the TX enable signal */
    - Co-authored-by: Kris Bahnsen <kris@embeddedTS.com>
  - turn off DMA for rs-485 (uart4)
  - usdhc1 as tested plus cleanup
  - usdhc2 cleanup
  - pinctrl node names should end in "grp"
  - num-cs for ecspi[34] do not need fsl
  - gpio2 is unused (remove node with all-empty gpio-line-names)
  - Preferred name for ts4900-gpio (i2c) is gpio-controller@28, not gpio@28
  - Identify gpio controllers asssociated with gpio-line-names (faster than looking up)
  - Remove trailing whitespace from all gpio-line-names sections
  - For the i2c gpio controller, specifying "base" is no longer needed nor allowed
  - For off-board SPI, remove spi-cs-high because it is not (and breaks in 5.10)
  - Number leds from 0 rather than following reference designators

markfeathers commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,768 @@
+		pinctrl-0 = <&pinctrl_gpio_leds>;
+		compatible = "gpio-leds";
+
+		led-2 {
```
Any reason to make these led-2 3 4 6  and not number from 0-3?

lionel replied:
These had been following the reference designators on the schematic; not so much needed with the way LEDs are now specified. Will renumber to 0

lionel committed:
ARM: dts: ts7180: Alphabetize as much as is reasonable, but not more

markfeathers commented:
Implemented tests and passed tests on the CI rack.

ts-kris commented on arch/arm/configs/tsimx6ul_defconfig:
```diff
@@ -676,6 +676,9 @@ CONFIG_SPI_GPIO=y
 CONFIG_SPI_IMX=y
 CONFIG_SPI_OCORES=m
 CONFIG_SPI_SPIDEV=y
+CONFIG_PPS=m
+CONFIG_PPS_CLIENT_LDISC=m
+CONFIG_PPS_CLIENT_GPIO=m
```
@markfeathers were these options added to the CI checks?

markfeathers replied:
Yes, we make sure pps0 is created on both kernel configs.

ts-kris commented on arch/arm/boot/dts/imx6ul-ts7180.dts:
```diff
@@ -0,0 +1,767 @@
+&uart4 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart4>;
+	uart-has-rtscts;
```
The `uart-has-rtscts` property is used to indicate the presence of hardware RTS/CTS in use. As this UART is using GPIO for RTS/TXEN, this property should be removed.

lionel replied:
Good catch of a legacy mistake. Thanks!
